src/app/operators/takeUntilDestroy.operator.ts

```diff
@@ -0,0 +1,11 @@
+import { ChangeDetectorRef, inject, ViewRef } from '@angular/core';
+import { Observable, Subject, takeUntil, UnaryFunction } from 'rxjs';
+
+export const takeUntilDestroy = <T>(): UnaryFunction<Observable<T>, Observable<T>> => {
+	const viewRef = inject(ChangeDetectorRef) as ViewRef;
+	const destroyer$ = new Subject<void>();
+
+	viewRef.onDestroy(() => destroyer$.next());
```

Comment:
takeUntil pattern typically calls complete to ensure any possible leak for the subject is done. Can we add destroyer$.complete()

Reply:
I added this in my [f17121c](https://github.com/bitovi/opensource-analytics-dashboard/pull/8/commits/f17121c7455ba1dfb5167e4c13ddec7e30b4a1ce) commit